corehq/apps/users/views/__init__.py

```diff
@@ -1069,6 +1054,27 @@ def post(self, request, *args, **kwargs):
         except WorksheetNotFound:
             self.group_specs = []
 
+
+class UploadWebUsers(BaseUploadUser):
+    template_name = 'hqwebapp/bulk_upload.html'
+    urlname = 'upload_web_users'
+    page_title = ugettext_noop("Bulk Upload Web Users")
+
+    @method_decorator(always_allow_project_access)
+    @method_decorator(require_can_edit_web_users)
+    @method_decorator(requires_privilege_with_fallback(privileges.BULK_USER_MANAGEMENT))
+    def dispatch(self, request, *args, **kwargs):
+        return super(UploadWebUsers, self).dispatch(request, *args, **kwargs)
+
+    @property
+    def page_context(self):
+        request_params = self.request.GET if self.request.method == 'GET' else self.request.POST
+        from corehq.apps.users.views.mobile import get_user_upload_context
+        return get_user_upload_context(self.domain, request_params, "download_web_users", "web user", "web users")
+
+    def post(self, request, *args, **kwargs):
```

Comment:
I think you could move the rest of this to the base class by adding a `is_web_user` attribute to the class, and passing that attribute in to the two functions that need it as an argument

Reply:
I think the parallel stuff might actually be extendable to both as well which might be nice for NY, but if that remains a difference, I think that is fine.

Reply:
added `is_web_upload` [here](https://github.com/dimagi/commcare-hq/pull/28832/commits/ddf9bdc627bc8bb869ca5282704dcab98a161ccf). I left the parallel out for now because I would like to do more testing for that than when is currently scoped out for this PR